File: easy/delete_node_linked_list/delete_node_linked_list.py
class Node(object):
    def __init__(self, val):
        self.val = val
        self.next = None

    def push(self, val):
        new_node = Node(val)
        curr = self
        while curr.next != None:
            curr = curr.next
        curr.next = new_node
        return

    def print(self):
        curr = self
        while curr != None:
            print(curr.val, end=" -> ")
            curr = curr.next
        print()
        return


class Solution(object):
    def deleteNode(self, node):
        """
        :type node: ListNode
        :rtype: void Do not return anything, modify node in-place instead.
        """
        if node.next != None:
            node.val = node.next.val
            node.next = node.next.next
        return


s = Solution()

print("Solution 1")
head = Node(4)
head.push(5)
head.push(1)
head.push(9)
head.print()
s.deleteNode(head.next)
head.print()

File: easy/delete_node_linked_list/delete_node_linked_list.go
package main

import "fmt"

type Node struct {
	Val  int
	Next *Node
}

func (n *Node) push(val int) {
	newNode := &Node{Val: val}
	curr := n
	for curr.Next != nil {
		curr = curr.Next
	}
	curr.Next = newNode
	return
}

func (n *Node) print() {
	curr := n
	for curr != nil {
		fmt.Print(curr.Val, " -> ")
		curr = curr.Next
	}
	fmt.Println()
}

func deleteNode(node *Node) {
	if node.Next != nil {
		node.Val = node.Next.Val
		node.Next = node.Next.Next
	}
	return
}

func main() {
	head := &Node{Val: 4}
	head.push(5)
	head.push(1)
	head.push(9)
	head.print()
	deleteNode(head.Next)
	head.print()
}
